dashboard/config/routes.rb

```diff
@@ -620,6 +620,7 @@
       get 'projects/gallery/public/:project_type/:limit(/:published_before)', to: 'projects/public_gallery#index', defaults: {format: 'json'}
 
       get 'projects/personal', to: 'projects/personal_projects#index', defaults: {format: 'json'}
+      get 'libraries/classmates/all', to: 'section_libraries#get_classmate_published_libraries', defaults: {format: 'json'}
```

Comment:
Please follow [rails conventions](https://guides.rubyonrails.org/routing.html#resource-routing-the-rails-default), as opposed to following the conventions you see in this file. In this case, please change this line to:
```
resources :section_libraries, only: [:index], defaults: {format: 'json'}
```
Then rename your `get_classmate_published_libraries` method to `index`, and have your client `GET /api/v1/section_libraries/`.